### PR TITLE
RE2022-330: Fix missing sample table entries bug

### PR DESCRIPTION
**Pull request by MrCreosote**
Not all sample entries in the DB have the same keys, which can mean the table view of the samples table may have out of order columns for samples with missing keys. Fill in missing values with `null`.

### CODE REVIEW

File: src/service/data_products/common_functions.py [hunk 1]
```diff
@@ -337,36 +340,31 @@ async def query_table(
...
 
     storage - the storage system.
+    columns - the expected columns in the table data.
     filters - the filters to apply to the search
     output_table - whether to return the results as a list of lists (e.g. a table) with a separate
         fields entry defining the key for each table column, or a list of key / value dictionaries.
     document_mutator - a function applied to a document retrieved from the database before
         returning the results.
     """
+    fields = [c.key for c in columns]
```
Comment (Tianhao-Gu) [inline src/service/data_products/common_functions.py:350]: should we enforce `sort` here rather than relying on caller to pass a `sorted(cols)`?

Comment (MrCreosote) [reply src/service/data_products/common_functions.py:350]: I'm assuming that in the future we'll probably want to define a sort order in the column specs, in which case the calling code will need to provide the sort order to `query_table`

Comment (Tianhao-Gu) [reply src/service/data_products/common_functions.py:350]: 👍

**Review by Tianhao-Gu — APPROVED** [review]
💯

**Pull request merged**